### ISSUE TITLE
NPE in FireballManager 

### ISSUE THREAD

**Issue by galacticwarrior9** [TylerS1066/Movecraft-Combat#35]
On the latest build of Tuinity 1.16.2, the following NPE is constantly thrown:

```
[20:07:22 WARN]: [Movecraft-Combat] Task #484 for Movecraft-Combat v1.0.11 generated an exception
java.lang.NullPointerException: null
        at net.countercraft.movecraft.combat.movecraftcombat.fireballs.FireballManager.run(FireballManager.java:29) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.b(MinecraftServer.java:1354) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at net.minecraft.server.v1_16_R2.DedicatedServer.b(DedicatedServer.java:372) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.a(MinecraftServer.java:1267) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:999) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.3.jar:git-Tuinity-"5df4a4f"]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

It is triggered by line 29 of the FireballManager class, which is: 

`while(System.currentTimeMillis() - q.peek().getMetadata("MCC-Expiry").get(0).asLong() > timeLimit && q.size() > 0)`

IntelliJ highlights `.getMetadata("MCC-Expiry")` as the cause, which I assume is because `q`, the LinkedList defined in line 22, can be empty?

**Comment by TylerS1066** [comment]
Have you verified this is not a bug with Tuinity and occurs in Paper or Spigot?  I've seen a lot of odd behavior with Tuinity and am not interested in tracking down something if it is due to their changes.

**Comment by galacticwarrior9** [comment]
Yes. I can verify it occurs on the latest builds of Paper (#246) and Spigot (git-Spigot-57bbdd8-55a78ff).

**Comment by cccm5** [comment]
https://github.com/TylerS1066/Movecraft-Combat/blob/468dd9a7e7eaaac09cd69dc50a3807d5722b7129/src/main/java/net/countercraft/movecraft/combat/movecraftcombat/fireballs/FireballManager.java#L29
the size check should be done first to short circuit the boolean statement and prevent peeking an empty queue.

**Comment by TylerS1066** [comment]
> Yes. I can verify it occurs on the latest builds of Paper (#246) and Spigot (git-Spigot-57bbdd8-55a78ff).

Thank you!



> the size check should be done first to short circuit the boolean statement and prevent peeking an empty queue.

I suspected as much but due to the fact that I had never seen this on my testing I wanted to verify it wasn't tuinity.

**Comment by TylerS1066** [comment]
This should be fixed in fb4e91c37087e605fdf4d2b33cba7af6639a4015, and you can get a beta build over at the [GitHub Actions Build](https://github.com/TylerS1066/Movecraft-Combat/actions/runs/339212353).
